### PR TITLE
Reworks (and nerfs) Hypernob internals for plasmamens

### PR DESCRIPTION
**Pull request by GuillaumePrata**
## About The Pull Request
Hypernob protection is now a status effect. It gives better feedback to the players on how it works since it is a clear buff icon(based on the other atmos/internal icons) on their Hud with an explanation when hovered over.
The protection status effect was also nerfed and gives a small slowdown.

hypernob_protection

## Why It's Good For The Game
My Dad always used to say: "There is only two things certain in life, death and powercreep."
When I made this PR: #52942 Hypernob was the true endgame of atmos so I felt it was a powerful but fair reward for atmos players.
Then this PR: #53835 changed how Hypernob was made and it became something quite simple to achieve.

This was a good change for atmos tho, and since plasmamen always had chemistry tricks they can use to stay suitless I ignored my distaste for how I implemented the buff and kept telling myself I should fix it eventually. So we are here now.

The Hypernob reaction is quite powerful and had no downsides, the other atmos gas internals all have something you must juggle if you want to keep them 24/7 so I added a slowdown into it.

Plasmamen have been powercreeped a lot over the years and went from the species I liked the most for how unique they played to the one that I have the most issues with, as they ignore a huge slice of the game balance, so I will take responsibility for the slice I added.

## Changelog

:cl: Guillaume Prata
qol: Hypernob Protection (given to plasmamen that are breathing Hypernob as gas) is a status effect now and gives better feedback to the user.
balance: Hypernob Protection gives a small slowdown.
/:cl:

### CODE REVIEW

**Comment by Colovorat** [comment]
S-sooo how does it nerfs plasmemes?

**Comment by Pickle-Coding** [comment]
Hypernob is pretty heavy, so it makes sense that it would slow you down after inhaling, as there is more mass to move.

**Comment by ATH1909** [comment]
slowdown effects really should not be handed out lightly, they absolutely cripple you in combat and make day to day traversal of the station painful

I... don't really see why a plasmaman would ever make/request a hypernobilium mix for themselves if it slows them down.

**Comment by TheBoondock** [comment]
Breathing hypernob makes plasman immune to all fire, it's quite powerful when combo with halon making you a walking terror with a trit flamethrower. It's quite sad to see it nerf

**Comment by GuillaumePrata** [comment]
> can anyone come up with a use case for an internals mix with this proposed hypernobilium change?

When they value the fireproof or suitless meme more than their move speed.
There is also 20 other ways to speed yourself up if you want to tip the scale, atmos has one already with nitrium.

**Review by tralezab — APPROVED** [review]
looks good. god, status_effect has some weird patterns in it.

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 1]
```diff
@@ -1500,15 +1500,10 @@
...
-	if(isplasmaman(L))
-		ADD_TRAIT(L, TRAIT_NOFIRE, type)
-
-/datum/reagent/hypernoblium/on_mob_end_metabolize(mob/living/L)
-	if(isplasmaman(L))
-		REMOVE_TRAIT(L, TRAIT_NOFIRE, type)
-	return ..()
+/datum/reagent/hypernoblium/on_mob_life(mob/living/carbon/M, delta_time, times_fired)
+	if(isplasmaman(M))
+		M.set_timed_status_effect(10 SECONDS * REM * delta_time, /datum/status_effect/hypernob_protection)
```
[the remark this answers is not in the document]
Comment (GuillaumePrata) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm:1505]: I will fix and test it in the morning then, thank you.

**Pull request merged**